### PR TITLE
Migrate .packages -> package_config.json

### LINKED ISSUE #99678: Flutter should not consume .packages
pub will stop producing .packages: https://github.com/dart-lang/pub/pull/2757

Instead rely on the replacement .dart_tool/package_config.json

Currently rolling the above into the sdk (https://dart-review.googlesource.com/c/sdk/+/231101) fails as such:
https://ci.chromium.org/ui/p/dart/builders/try/flutter-engine-linux-try/678/overview

### PR DESCRIPTION
**Pull request by sigurdm**
Replace .packages with .dart_tool/package_config.json in BuildInfo.packagesPath.

Fixes: https://github.com/flutter/flutter/issues/99678

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by sigurdm** [comment]
Not sure about the failing windows test - is it a regular timeout or is something else going on...

**Review by jonahwilliams — APPROVED** [review]
I think the test itself might be flaky right now, though @christopherfujino might know more.

Otherwise LGTM, if LG to @christopherfujino

**Comment by christopherfujino** [comment]
Re-ran the tool integration test.

File: packages/flutter_tools/test/general.shard/flutter_tester_device_test.dart
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/flutter_tester_device_test.dart:289]: because this is now the default argument

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/flutter_tester_device_test.dart:289]: ahh, cool

File: packages/flutter_tools/lib/src/build_info.dart
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_info.dart:39]: Maybe this hard-coded path separator is leading to coverage collection failing on Windows?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_info.dart:39]: odd that it is failing only for windows. Really, you should be able to use `/` conditionally ...

Perhaps the coverage code is doing some string matching instead of using package:path ?

Comment (sigurdm) [reply packages/flutter_tools/lib/src/build_info.dart:39]: Using the platform separator will make the constructor non-const (which might not be a problem).

Cursory inspection doesn't show anything in package:coverage that should depend on backslash on windows...

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_info.dart:39]: I'll debug this one today, I use a windows machine

Comment (sigurdm) [reply packages/flutter_tools/lib/src/build_info.dart:39]: Gentle ping

This is blocking https://dart-review.googlesource.com/c/sdk/+/231101 .

**Comment by jonahwilliams** [comment]
Fix is https://github.com/dart-lang/coverage/pull/371 , which needs to be rolled into flutter first

**Comment by christopherfujino** [comment]
> Fix is [dart-lang/coverage#371](https://github.com/dart-lang/coverage/pull/371) , which needs to be rolled into flutter first

That PR was merged, we are now blocked on a new published pub version. @bkonyi or @natebosch do you know the process for publishing `package:coverage`? it looks like there are several batched changes.

**Comment by bkonyi** [comment]
> > Fix is [dart-lang/coverage#371](https://github.com/dart-lang/coverage/pull/371) , which needs to be rolled into flutter first
> 
> That PR was merged, we are now blocked on a new published pub version. @bkonyi or @natebosch do you know the process for publishing `package:coverage`? it looks like there are several batched changes.

Either @natebosch or @liamappelbe (maybe) should be able to publish a new version.

**Comment by christopherfujino** [comment]
> > > Fix is [dart-lang/coverage#371](https://github.com/dart-lang/coverage/pull/371) , which needs to be rolled into flutter first
> > 
> > 
> > That PR was merged, we are now blocked on a new published pub version. @bkonyi or @natebosch do you know the process for publishing `package:coverage`? it looks like there are several batched changes.
> 
> Either @natebosch or @liamappelbe (maybe) should be able to publish a new version.

Pretty please :)

**Comment by liamappelbe** [comment]
> > > > Fix is [dart-lang/coverage#371](https://github.com/dart-lang/coverage/pull/371) , which needs to be rolled into flutter first
> > > 
> > > 
> > > That PR was merged, we are now blocked on a new published pub version. @bkonyi or @natebosch do you know the process for publishing `package:coverage`? it looks like there are several batched changes.
> > 
> > 
> > Either @natebosch or @liamappelbe (maybe) should be able to publish a new version.
> 
> Pretty please :)

Done

**Comment by christopherfujino** [comment]
@sigurdm if you rebase to https://github.com/flutter/flutter/pull/100726, you should pick up the fix to `package:coverage`

**Pull request merged**
